src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -190,20 +188,41 @@ protected void buildEndpoint(ApiEndpoint baseEndpoint) {
     private ResponseList<T> getMany(Request req, Response res) {
         int limit = HttpUtils.getQueryParamFromRequest(req, LIMIT_PARAM, 0, DEFAULT_LIMIT, 100);
         int offset = HttpUtils.getQueryParamFromRequest(req, OFFSET_PARAM, 0, DEFAULT_OFFSET);
-        Auth0UserProfile requestingUser = getUserFromRequest(req);
-        if (isUserAdmin(requestingUser)) {
+        String userId = HttpUtils.getQueryParamFromRequest(req, USER_ID_PARAM, true);
+        // Filter the response based on the user id, if provided.
+        // If the user id is not provided filter response based on requesting user.
+        RequestingUser requestingUser = Auth0Connection.getUserFromRequest(req);
+        if (userId != null) {
+            OtpUser otpUser = Persistence.otpUsers.getById(userId);
+            if (otpUser != null && otpUser.canBeManagedBy(requestingUser)) {
```

Comment:
This conditional is used multiple times. Maybe it should be a convenience method on `RequestingUser` (e.g., `RequestingUser#canManageEntity`).

Reply:
Updated.